include/alp/algorithms/cholesky.hpp

```diff
@@ -55,54 +60,128 @@ namespace alp {
 			const size_t n = nrows( H );
 
 			// Out of place specification of the operation
-			Matrix< D, structures::SymmetricPositiveDefinite, Dense > LL( n, n );
+			Matrix< D, structures::Symmetric, Dense > LL( n, n );
 			rc = set( LL, H );
+#ifdef DEBUG
+			if( rc != SUCCESS ) {
+				std::cerr << " set( LL, H ) failed\n";
+				return rc;
+			}
+			print_matrix( " -- LL --  " , LL );
+#endif
 
 			for( size_t k = 0; k < n ; ++k ) {
+#ifdef DEBUG
+				std::cout << "============ Iteration " << k << " ============" << std::endl;
+#endif
 
-				const size_t m = n - k - 1;
-
-				auto v = get_view( LL , utils::range( k, n) , k );
+				auto a = get_view( LL, k, utils::range( k, n ) );
+#ifdef DEBUG
+				print_vector( " -- a --  " , a );
+#endif
 
 				// L[ k, k ] = alpha = sqrt( LL[ k, k ] )
 				Scalar< D > alpha;
 				rc = eWiseLambda(
-							[ &v, &alpha, &ring ]( const size_t i ) {
-								if ( i == 0 ) {
-									v[ i ] = alpha = std::sqrt( v[ i ] );
-								}
-							},
-							v );
-
+					[ &alpha, &ring ]( const size_t i, D &val ) {
+						if ( i == 0 ) {
+#ifdef TEMP_DISABLE
+							internal::setInitialized( alpha, true );
+							*alpha = std::sqrt( val );
+#else
+							(void)set( alpha, std::sqrt( val ) );
+#endif
+							val = *alpha;
+						}
+					}, a
+				);
+
+#ifdef DEBUG
+				std::cout << "alpha " << *alpha << std::endl;
+				if( rc != SUCCESS ) {
+					std::cerr << " eWiseLambda( lambda, view ) (0) failed\n";
+					return rc;
+				}
+#endif
+
+				auto v = get_view( LL, k, utils::range( k + 1, n ) );
+#ifdef DEBUG
+				print_vector( " -- v --  " , v );
+#endif
 				// LL[ k + 1: , k ] = LL[ k + 1: , k ] / alpha
 				rc = eWiseLambda(
-							[ &v, &alpha, &divide ]( const size_t i ) {
-								if ( i > 0 ) {
-									foldl( v[ i ], alpha, divide );
-								}
-							},
-							v );
-
+					[ &alpha, &divide ]( const size_t i, D &val ) {
+						(void)i;
+						internal::foldl( val, *alpha, divide );
+					},
+					v
+				);
+
+#ifdef DEBUG
+				if( rc != SUCCESS ) {
+					std::cerr << " eWiseLambda( lambda, view ) (1) failed\n";
+					return rc;
+				}
+#endif
 
 				// LL[ k+1: , k+1: ] -= v*v^T
 				auto LLprim = get_view( LL, utils::range( k + 1, n ), utils::range( k + 1, n ) );
 
-				vvt = outer( v, ring.getMultiplicativeOperator() );
+				auto vvt = outer( v, ring.getMultiplicativeOperator() );
+#ifdef DEBUG
+				print_vector( " -- v --  " , v );
+				print_matrix( " vvt ", vvt );
+#endif
+#ifdef TEMP_DISABLE
+				rc = alp::eWiseLambda(
+					[ &vvt, &minus, &divide ]( const size_t i, const size_t j, D &val ) {
+						internal::foldl(
+							val,
+							internal::access( vvt, internal::getStorageIndex( vvt, i, j ) ),
+							minus
+						);
+#ifdef DEBUG
+						std::cout << "lambda " << i << " " << j << " " << val
+							  << " " << internal::access( vvt, internal::getStorageIndex( vvt, i, j ) ) << std::endl;
+#endif
```

Comment:
I think we could remove this because it is rather low-level and is not checked in other similar calls.